### PR TITLE
🛠️ Refactor daycare tests

### LINKED ISSUE #72: Refactoring Daycare tests
## Expected Behavior

Improving the implementation of using static data of daycares in the daycare tests app

## Current Behavior

Using constants in 'tests_daycare.py' file

## Acceptance Criteria

remove the need for the constants and make the tests better by using `pytest.mark.parametrize`

### Steps to Reproduce

- [ ] Remove constans from 'tests_daycare.py' file
- [ ] Add  data for testing the daycare app by using `pytest.mark.parametrize`
- [ ] Pass all the tests by pytest

Depends #60

### PR DESCRIPTION
**Pull request by ErezCohenn**
### Desired Outcome

Improving the implementation of using static data invalidations tests of daycares in the daycare app

### Implemented Changes

By using `pytest.mark.parametrize` instead of constants, and reduction the all invalidations tests to one test

### Connected Issue/Story

Resolves #72 
Depends #60 

### Definition of Done
- [x] Remove constans from 'tests_daycare.py' file
- [x] Add  data for testing the daycare app by using `pytest.mark.parametrize`
- [x] Pass all the tests by pytest

#### Changelog

- [ ] The CHANGELOG has been updated
- [x] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [x] This PR includes new unit and integration tests to go with the code
  changes
- [ ] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [x] This PR does not require updating any documentation

### CODE REVIEW

File: daycare/test_daycare.py [hunk 1]
```diff
@@ -37,72 +25,43 @@ def test_daycare_user_delete(self, create_daycare_user):
...
+             'Invalid price - max price is 4 digits.'),
+            ('test@gmail.com', 'testuser02', 'test_password', 'test_name', 'test description',
+             10, 50, 'this is very long name above 20 letters',
+             'tel-aviv', 'test address', 'Invalid area - max length is 20'),
+            ('test@gmail.com', 'testuser02', 'test_password', 'test_name', 'test description',
+             10, 50, 'Merkaz', 'this is very long name above 20 letters', 'test address',
+             'Invalid city - max length is 20'),
+            ('test@gmail.com', 'testuser02', 'test_password', 'test_name', 'test description',
+             10, 50, 'Merkaz', 'tel-aviv', 'this is very long name above 50 letters and maybe moreeeeeee',
+             'Invalid address - max length is 50')
```
[the remark this answers is not in the document]
Comment (ErezCohenn) [reply daycare/test_daycare.py]: Done

File: daycare/test_daycare.py [hunk 2]
```diff
@@ -37,72 +25,39 @@ def test_daycare_user_delete(self, create_daycare_user):
...
         assert create_daycare_user not in DayCare.objects.all()
 
-    def test_daycare_user_creation_with_invalid_email(self):
-        with pytest.raises(ValidationError, match="'Enter a valid email address.'"):
-            DayCare.create(username='testuser02', email="invalid mail format", password=PASSWORD,
-                           name=NAME, description=DESCRIPTION,
-                           price_per_day=PRICE_PER_DAY,
-                           capacity=CAPACITY, area=AREA, city=CITY, address=ADDRESS)
+    @pytest.mark.parametrize(
+        ['email', 'name', 'price_per_day', 'capacity',
```
Comment (Omeramsc) [inline daycare/test_daycare.py:29]: no need for capacity

File: daycare/test_daycare.py [hunk 3]
```diff
@@ -37,72 +25,39 @@ def test_daycare_user_delete(self, create_daycare_user):
...
+            pytest.param('test@gmail.com', 'test_name', 10000, 50, 'Merkaz', 'tel-aviv', 'test address',
+                         'Invalid price - max price is 4 digits.'),
+            pytest.param('test@gmail.com', 'test_name', 10, 50, 'this is very long name above 20 letters', 'tel-aviv',
+                         'test address', 'Invalid area - max length is 20'),
+            pytest.param('test@gmail.com', 'test_name', 10, 50, 'Merkaz', 'this is very long name above 20 letters',
+                         'test address', 'Invalid city - max length is 20'),
+            pytest.param('test@gmail.com', 'test_name', 10, 50, 'Merkaz', 'tel-aviv',
+                         'this is very long name above 50 letters and maybe moreeeeeee',
+                         'Invalid address - max length is 50')
+        ],
```
Comment (Omeramsc) [inline daycare/test_daycare.py:47]: I'd suggest adding an `id="test description"` field to each `pytest.param`. you don't have to include it in the params list.

**Review by Omeramsc — CHANGES_REQUESTED** [review]
LGTM, fix the minor comments and it can be merged IMO

File: daycare/test_daycare.py (hunk 2 above)
Comment (ErezCohenn) [reply daycare/test_daycare.py]: Nice catch! done :)

File: daycare/test_daycare.py (hunk 3 above)
Comment (ErezCohenn) [reply daycare/test_daycare.py:47]: done

**Review by Omeramsc — COMMENTED** [review]
LGTM

**Pull request merged**